src/System.Management.Automation/engine/parser/TypeInferenceVisitor.cs

```diff
@@ -1099,10 +1292,78 @@ private void InferTypesFromPreviousCommand(CommandAst commandAst, List<PSTypeNam
 
         private void InferTypesFromSelectCommand(PseudoBindingInfo pseudoBinding, CommandAst commandAst, List<PSTypeName> inferredTypes)
         {
-            if (pseudoBinding.BoundArguments.TryGetValue("Property", out _)
-                || pseudoBinding.BoundArguments.TryGetValue("ExcludeProperty", out _))
+            void InferFromSelectProperties(AstParameterArgumentPair astParameterArgumentPair, CommandBaseAst previousPipelineElementAst, bool includeMatchedProperties = true)
             {
-                return;
+                if (astParameterArgumentPair is AstPair astPair)
+                {
+                    object ToWildCardOrString(string value) => WildcardPattern.ContainsWildcardCharacters(value) ? (object)new WildcardPattern(value) : value;
```

Comment:
Consider using PSPropertyExpression - it already handles wildcards in property names.

Reply:
Maybe even use the ParameterProcessor? So we can handle hashtable arguments also?
I'll leave that to a future PR.